### PR TITLE
Added UUID arrray codec

### PR DESCRIPTION
**Pull request by richlackey**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/pgjdbc/r2dbc-postgresql/blob/main/.github/CONTRIBUTING.adoc).
- [x] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [x] You use the code formatters provided [here](https://github.com/pgjdbc/r2dbc-postgresql/blob/master/intellij-style.xml) and have them applied to your changes. Don't submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

### CODE REVIEW

File: src/main/java/io/r2dbc/postgresql/codec/UuidArrayCodec.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import io.r2dbc.postgresql.message.Format;
+import io.r2dbc.postgresql.type.PostgresqlObjectId;
+import io.r2dbc.postgresql.client.Parameter;
+import io.r2dbc.postgresql.util.Assert;
+import reactor.util.annotation.Nullable;
+
+import java.util.UUID;
+import java.util.function.Supplier;
+
+public class UuidArrayCodec extends AbstractArrayCodec<UUID> {
```
Comment (mp911de) [inline src/main/java/io/r2dbc/postgresql/codec/UuidArrayCodec.java:14]: Nit: codecs are typically package-private and `final`.

File: src/main/java/io/r2dbc/postgresql/codec/UuidArrayCodec.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package io.r2dbc.postgresql.codec;
```
Comment (mp911de) [inline src/main/java/io/r2dbc/postgresql/codec/UuidArrayCodec.java:1]: Care to add a license header (see other files).

File: README.md [hunk 3]
```diff
@@ -394,6 +394,7 @@ Support for the following single-dimensional arrays (read and write):
 |:-----------------------------------------------|:-------------------------------------|
 |[`text[]`][psql-text-ref]                        |[`String[]`][java-string-ref]         |  
 |[`integer[] or int[]`][psql-integer-ref]        |[`Integer[]`][java-integer-ref], [`Long[]`][java-long-ref], [`Short[]`][java-short-ref]|
+|[`uuid[]`][psql-uuid-ref]        |[`UUID[]`][java-uuid-ref]|
```
Comment (mp911de) [inline README.md:397]: Nit: Formatting

**Review by mp911de — COMMENTED** [review]
Great work. I left a few comments about things that would be nice to sort out before the merge. Also, can you extend `AbstractCodecIntegrationTests` by adding an integration test for UUID arrays?

File: src/main/java/io/r2dbc/postgresql/codec/UuidArrayCodec.java (hunk 1 above)
Comment (richlackey) [reply src/main/java/io/r2dbc/postgresql/codec/UuidArrayCodec.java:14]: Ok

Comment (richlackey) [reply src/main/java/io/r2dbc/postgresql/codec/UuidArrayCodec.java]: Flipped to local final

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged, polished, and backported now.

**Pull request closed without merging**